VERA/ZSM: remember last sample played so that async pitch changes work on ZSM export

mooinglemur opened the pull request:
Prior to this PR, doing slides/portamento/E5xx effects/legato on the VERA PCM channel would export the incorrect pitch because of the way ZSM export works.  PCM events on ZSM export are consumed on the tick that they start, so on export, pitch changes did not know which sample was playing and the divisor ended up being the default 65536.

Remembering the last `s->centerRate` and using its value if there's no sample playing seems like a workable fix, but I'm not sure if it's the best fix.

tildearrow reviewed (changes requested):
Wait......... lastCenterRate should be set to -1 on reset()... right?

mooinglemur commented:
Yup, you're right. Moved it out of the constructor.  I also moved the setting of it outside of the conditional.  If a sample is playing and its centerRate value out of bounds, we should store it, as it _was_ the last value.
